### PR TITLE
include: Add i2c helper methods for async implementation

### PR DESCRIPTION
**Pull request by hernangtz**
Added 3 helper methods for async calls as they exist for the blocking implementation

Signed-off-by: Hernan I. Gutierrez-Vazquez <quic_hernang@quicinc.com>

### CODE REVIEW

**Comment by teburd** [comment]
This needs to be rebased, it looks like a merge commit was added and this is the wrong way to update this PR. Please rebase your i2c changes on to main and force push, there should be no merges/merge commits.

**Comment by hernangtz** [comment]
> This needs to be rebased, it looks like a merge commit was added and this is the wrong way to update this PR. Please rebase your i2c changes on to main and force push, there should be no merges/merge commits.

yes. I used the button in the gui to rebase and it added an extra commit, I will fix it.

**Review by teburd — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Comment by hernangtz** [comment]
@carlescufi  I just rebased and all test have passed, would you mind taking a look at this PR?
@teburd have approved also after I rebased and repushed.

**Comment by hernangtz** [comment]
> @teburd please take another look

I had to rebase/repush as branch went behind again.

**Review by teburd — DISMISSED** [review]

File: include/zephyr/drivers/i2c.h [hunk 1]
```diff
@@ -690,6 +690,103 @@ static inline int i2c_transfer_cb(const struct device *dev,
...
+ *
+ * @param spec I2C specification from devicetree.
+ * @param msgs Array of messages to transfer.
+ * @param num_msgs Number of messages to transfer.
+ * @param cb Function pointer for completion callback.
+ * @param userdata Userdata passed to callback.
+ *
+ * @return a value from i2c_transfer_cb()
+ */
+static inline int i2c_transfer_dt_cb(const struct i2c_dt_spec *spec,
```
Comment (MaureenHelm) [inline include/zephyr/drivers/i2c.h:708]: This should be `i2c_transfer_cb_dt`

File: include/zephyr/drivers/i2c.h [hunk 2]
```diff
@@ -690,6 +690,103 @@ static inline int i2c_transfer_cb(const struct device *dev,
...
+ * @param write_buf Pointer to the data to be written
+ * @param num_write Number of bytes to write
+ * @param read_buf Pointer to storage for read data
+ * @param num_read Number of bytes to read
+ * @param cb Function pointer for completion callback.
+ * @param userdata Userdata passed to callback.
+ *
+ * @return a value from i2c_write_read_cb()
+ */
+static inline int i2c_write_read_dt_cb(const struct i2c_dt_spec *spec, struct i2c_msg *msgs,
```
Comment (MaureenHelm) [inline include/zephyr/drivers/i2c.h:781]: This should be `i2c_write_read_cb_dt`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: include/zephyr/drivers/i2c.h (hunk 1 above)
Comment (hernangtz) [reply include/zephyr/drivers/i2c.h:708]: ok, will fix in few

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Comment by teburd** [comment]
@MaureenHelm good catches

**Pull request merged**